test/functional/tests/misc/test_device_capabilities.py

```diff
@@ -183,11 +183,23 @@ def compare_capabilities(cache_device, core_device, cache, core, msg):
             cache_val = cache_dev_capabilities[capability]
             core_val = core_dev_capabilities[capability]
 
-            comparison_val = method(core_val, cache_val) if method is not None else core_val
-
-            if comparison_val != cas_val:
+            expected_val = method(core_val, cache_val) if method is not None else core_val
+
+            if capability in ["max_sectors_kb", "max_hw_sectors_kb"] and expected_val != cas_val:
+                # On the newer kernels this trait is rounded. Instead of checking for
+                # the current kernel version, assume that both values are acceptable.
```

Comment:
Wouldn't it introduce the risk for a pass in case of a justified mismatch? What I mean is - is it possible that _cas_val_ will be set to wrong value, which is equal to rounded _expected_val_?

Reply:
Yes, I'm aware of that but I consider it highly unlikely. Additionally, I believe this test main purpose is to verify if traits are inherited properly (either from cache or core device, depend on a particular trait), not to check the values themselves. Also, this logic with accepting two values may be removed as soon as we drop support for kernels below 5.10 :smile: